internals/plan/plan.go

```diff
@@ -811,11 +875,24 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 
 	// Validate service log targets
 	for serviceName, service := range combined.Services {
-		for _, targetName := range service.LogTargets {
-			_, ok := combined.LogTargets[targetName]
-			if !ok {
-				return nil, &FormatError{
-					Message: fmt.Sprintf(`unknown log target %q for service %q`, targetName, serviceName),
+		if service.LogTargets != nil && service.LogTargetsReplace != nil {
+			return nil, &FormatError{
+				Message: fmt.Sprintf(`service %q cannot define both "log-targets" and "^log-targets" keys`, serviceName),
+			}
+		}
+
+		logTargets := service.LogTargets
+		if logTargets == nil && service.LogTargetsReplace != nil {
+			logTargets = service.LogTargetsReplace.Targets
+		}
+
+		if logTargets != nil {
```

Comment:
Nit: probably don't need this `if` statement, as the range loop will do no iterations if `logTargets` is nil.

Reply:
nice catch!